Added new useful methods to Pokemon.java

gionata-bisciari opened the pull request:
+getStaminaInPercentage():int
+getCPInPercentageActualPlayerLevel():int
+getCPInPercentageMaxPlayerLevel:int
+getIvInPercentage():double

gionata-bisciari committed:
Added canEvolve()

+Added canEvolve():boolean to Pokemon.java

gionata-bisciari committed:
Merge remote-tracking branch 'upstream/Development' into Development

Update from Grover-c13/PokeGOAPI-Java

gionata-bisciari committed:
Merge remote-tracking branch 'upstream/Development' into Development

Merge updates from Grover-c13/PokeGOAPI-Java

gionata-bisciari committed:
Added new useful methods to Pokemon.java

+getStaminaInPercentage():int
+getCPInPercentageActualPlayerLevel():int
+getCPInPercentageMaxPlayerLevel():int
+getIvInPercentage():double

FabianTerhorst commented on library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:
```diff
@@ -403,4 +404,36 @@ public boolean isFainted() {
+	/**
+	 * @return IV in percentage
+	 */
+	public double getIvInPercentage(){
```
Add Space between ) and {

gionata-bisciari committed:
Brackets changed as suggested from @FabianTerhorst

Brackets changed as suggested from @FabianTerhorst

AbandonedCart commented:
That's a nice way to interpret those values, especially when comparing them.
Too bad at anything below 35%, it can't just return `NAME fails as a Pokemon`

Out of curiosity, why does the final one return a double?
As far as I know, Math.floor will already return an int.

gionata-bisciari commented:
![ivs](https://cloud.githubusercontent.com/assets/15786737/17710424/7b98f396-63ed-11e6-83da-5f29d947aabf.jpg)
it returns a double with two decimal places

AbandonedCart commented:
My mistake. Missed the second outer parenthesis for the / 100

FabianTerhorst commented:
Please fix the style warnings.

gionata-bisciari commented:
what style warnings?

FabianTerhorst commented:
When you run the gradle task on your project.

FabianTerhorst commented:
I still don´t know why travis doesn´t print them.

gionata-bisciari commented:
I haven't seen it infact, I'll check

gionata-bisciari committed:
Fixed style warnings

:library:extractIncludeProto UP-TO-DATE
:library:extractProto UP-TO-DATE
:library:generateProto UP-TO-DATE
:library:compileJava UP-TO-DATE
:library:processResources UP-TO-DATE
:library:classes UP-TO-DATE
:library:jar UP-TO-DATE
:library:bundle UP-TO-DATE
:library:javadoc UP-TO-DATE
:library:javadocJar UP-TO-DATE
:library:sourcesJar UP-TO-DATE
:library:assemble UP-TO-DATE
:library:checkstyleMain UP-TO-DATE
:library:extractIncludeTestProto UP-TO-DATE
:library:extractTestProto UP-TO-DATE
:library:generateTestProto UP-TO-DATE
:library:compileTestJava UP-TO-DATE
:library:processTestResources UP-TO-DATE
:library:testClasses UP-TO-DATE
:library:checkstyleTest UP-TO-DATE
:library:test UP-TO-DATE
:library:check UP-TO-DATE
:library:build UP-TO-DATE

FabianTerhorst commented:
👍

Grover-c13 commented:
yeah i really wish we could get travis to fail on checkstyle :(

gionata-bisciari commented:
Is now everything ok?
